examples/solids/src/setupdm.c

```diff
@@ -47,8 +47,10 @@ PetscErrorCode CreateDistributedDM(MPI_Comm comm, AppCtx app_ctx, DM *dm) {
   PetscFunctionBeginUser;
 
   if (!*filename) {
-    PetscInt dim = 3;
-    ierr = DMPlexCreateBoxMesh(comm, dim, PETSC_FALSE, NULL, NULL,
+    PetscInt dim = 3, faces[3] = {3, 3, 3};
+    ierr = PetscOptionsGetIntArray(NULL, NULL, "-dm_plex_box_faces",
+                                   faces, &dim, NULL); CHKERRQ(ierr);
+    ierr = DMPlexCreateBoxMesh(comm, dim, PETSC_FALSE, faces, NULL,
```

Comment:
I'm not sure I understand this change. I thought that DMPlex automatically picked up on the number of box faces for us.

Reply:
Not any more on PETSc `main`. When we require main (petsc-3.16), we can rip out dozens of lines from all the examples because every mesh can be read or generated (plus partitioned/distributed/refined/extruded) from the same 3 lines.
```c
DMCreate(comm, &dm);
DMSetType(dm, DMPLEX);
DMSetFromOptions(dm);
```

Reply:
Ah, so should we add this to the other PETSc examples as well?

Reply:
Yeah, I guess that could be this PR, but I figured we would squash-merge this whole PR and I needed this to run tests with my current PETSc. I don't want to start refactoring until we have representative tests and this PR is merged.

Reply:
And upcoming libCEED 0.10 could perhaps sync with PETSc 3.16 for the examples, which would unify a lot of mesh handling (which is just extraneous complexity for someone interested in learning how to build an application with libCEED).